### PR TITLE
Mccalluc/just get the first line

### LINKED ISSUE #8: Option to just get headers
heatmap-scatter-dash is currently doing two passes in app_runner_refinery, but the first time we really just want the column headers. For large files in particular, it'd be great if we didn't need to read the whole file.

### PR DESCRIPTION
**Pull request by mccalluc**
- Fix #8: Option to read just first row.
- Add tests which essentially duplicate doc tests, but are more precise.
- Rename methods in tests for clarity.

### CODE REVIEW

File: tests/test_dataframer.py [hunk 1]
```diff
@@ -9,16 +9,68 @@
...
+                [1, 2, 3]],
+                columns=['a', 'b', 'c'],
+                index=[0]
+            ),
+            kwargs={'col_zero_index': False})
+
+    def test_keep_strings(self):
+        self.assert_good_parse(
+            b',b,c,xxx\n1,2,3,X!',
+            pandas.DataFrame([
```
Comment (jkmarx) [inline tests/test_dataframer.py]: fix spacing, [[ ]] on one line, also below

File: README.md [hunk 2]
```diff
@@ -74,6 +74,19 @@ True
 
 ```
 
+If you don't want to read the whole file, but instead only the first
+row so you have information on the columns:
```
Comment (jkmarx) [inline README.md]: "If you don't want to read the whole file, but instead only the first
row so you have information on the columns:"

Can you fix this sentence for clarity?

**Review by jkmarx — APPROVED** [review]
Please read/fix comments.

File: dataframer/dataframer.py [hunk 3]
```diff
@@ -45,27 +48,18 @@ def parse(file, col_zero_index=True, keep_strings=False, relabel=False):
     :return: DataFrameInfo, which contains the DataFrame itself,
     and a dict of labels for the rows, if relabel is True
     '''
-
-    # The pandas default behavior is to look at filename extensions,
-    # but we decided we can't rely on those to be accurate.
-    compression_type = {
-        b'\x1f\x8b': 'gzip',
-        b'\x50\x4b': 'zip'
```
Comment (scottx611x) [inline dataframer/dataframer.py:53]: I forget if we cared about `.zips` or not

Comment (mccalluc) [reply dataframer/dataframer.py:53]: After talking with Geoff, I believe not... but if we ever would want to bring it back, I filed #7.
